Network/DNS/Transport.hs

```diff
@@ -210,10 +220,8 @@ tcpLookup gen ai q tm ctls =
             sendVC vc qry
             receiveVC vc
         case mres of
-            Nothing                     -> E.throwIO TimeoutExpired
-            Just res
-                | checkResp q ident res -> return res
-                | otherwise             -> E.throwIO SequenceNumberMismatch
+            Nothing  -> E.throwIO TimeoutExpired
+            Just res -> maybe (return res) E.throwIO (checkRespM (Just q) ident res)
```

Comment:
There's no reason to change this code.  AXFR will use a different code path, because it needs to return a stream of resource records, with an IO interface to retrieve results that don't fit in a single 64k buffer.

Reply:
This change is unrelated to AXFR (modulo the `Just` that I will remove). This change ensures that sequence number and question mismatches do not get conflated under `SequenceNumberMismatch`.